test/over_react/util/equality_test.dart

```diff
@@ -0,0 +1,151 @@
+@TestOn('browser || vm')
+import 'package:over_react/src/util/equality.dart';
+import 'package:test/test.dart';
+
+main() {
+  group('areMapsShallowIdentical', () {
+    /// Expect that `areMapsShallowIdentical(a, b)` and `areMapsShallowIdentical(b, a)`
+    /// have the same result (represented by [matcher]).
+    void expectAreMapsShallowIdenticalCommutatively(Map a, Map b, matcher) {
+      final abResult = areMapsShallowIdentical(a, b);
+      final baResult = areMapsShallowIdentical(b, a);
+      expect([abResult, baResult], everyElement(matcher));
+    }
+
+    group('returns true for', () {
+      test('different maps that are equal', () {
+        final a = {
+          1: 'one',
+          2: 'two',
+        };
+        final b = {
+          1: 'one',
+          2: 'two',
+        };
+        expectAreMapsShallowIdenticalCommutatively(a, b, isTrue);
+      });
+
+      test('different maps that are equal with keys in different orders', () {
+        final a = {
+          1: 'one',
+          2: 'two',
+        };
+        final b = {
+          2: 'two',
+          1: 'one',
+        };
+        expectAreMapsShallowIdenticalCommutatively(a, b, isTrue);
+      });
+
+      test('different empty maps', () {
+        final a = {};
+        final b = {};
+        expectAreMapsShallowIdenticalCommutatively(a, b, isTrue);
+      });
+
+      test('the same map instance', () {
+        final a = {1: 'one', 2: 'two'};
+        final b = a;
+        expectAreMapsShallowIdenticalCommutatively(a, b, isTrue);
+      });
+    });
+
+    group('returns false for', () {
+      test('different maps that are equal', () {
+        final a = {
+          1: 'one',
+          2: 'two',
+        };
+        final b = {
+          2: 'two',
+          1: 'one',
+        };
+        expectAreMapsShallowIdenticalCommutatively(a, b, isTrue);
+      });
+
+      test('different empty maps', () {
+        final a = {};
+        final b = {};
+        expectAreMapsShallowIdenticalCommutatively(a, b, isTrue);
+      });
+
+      test('the same map instance', () {
+        final a = {1: 'one', 2: 'two'};
+        final b = a;
+        expectAreMapsShallowIdenticalCommutatively(a, b, isTrue);
+      });
+    });
```

Comment:
All the expectations in this section are for true return values... is this copypasta? 🍝

Reply:
Whoops, yes

Reply:
Addressed in 93fb99c173be4f1da9f8630d34680cda808774b5